src/pages/index.js

```diff
@@ -59,25 +145,83 @@ const newCardPopup = new PopupWithForm(".popup-card", (formValues) => {
     link: formValues["second-input"],
     alt: formValues["first-input"],
   };
-  const newCard = createCard(cardData);
-
-  section.addItem(newCard, "prepend");
-  newCardPopup.close();
-  addFormValidator.resetForm();
-  addFormValidator.disableSubmitButton();
+  const cardPopupButton = newCardPopup._submitButton;
+  api
+    .addCard(cardData)
+    .then((cardData) => {
+      const newCard = createCard(cardData);
+      section.addItem(newCard, "prepend");
+      newCardPopup.close();
+      addFormValidator.resetForm();
+      addFormValidator.disableSubmitButton();
+      // card creation here
+    })
+    .catch((err) => {
+      cardPopupButton.textContent = "ERROR";
+      console.error(err);
+    })
+    .finally(() => {
+      cardPopupButton.textContent = "Save";
+    });
+  cardPopupButton.textContent = "Saving...";
 });
 newCardPopup.setEventListeners();
 
+const confirmationPopup = new PopupWithSubmit(".popup-confirm");
+
+confirmationPopup.setEventListeners();
+
 /* ------------------------------ Profile Popup ----------------------------- */
 const newProfilePopup = new PopupWithForm(".popup-profile", (formValues) => {
   userInfo.setUserInfo({
     name: formValues["first-input"],
     description: formValues["second-input"],
   });
-  newProfilePopup.close();
+  api.updateProfileInformation({
+    name: formValues["first-input"],
+    about: formValues["second-input"],
+  })
+  .then(() => {
+  })
+  .catch((err) => {
+    popupButtonText.textContent = "ERROR";
+    console.error(err);
+  })
+  .finally(() => {
+    newProfilePopup.close();
```

Comment:
`    newProfilePopup.close();
`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**